iceoryx_posh/include/iceoryx_posh/capro/service_description.hpp

```diff
@@ -29,16 +30,12 @@ namespace iox
 {
 namespace capro
 {
-static constexpr uint16_t InvalidID = 0u;
-static const IdString_t InvalidIDString{"0"};
-static constexpr uint16_t AnyService = 0xFFFFu;
-static constexpr uint16_t AnyInstance = 0xFFFFu;
-static constexpr uint16_t AnyEvent = 0xFFFFu;
+static const IdString_t InvalidString{"INVALID"};
```

Comment:
Could this maybe be something more like `#$%&@_INVALID_@&%$#` or something similar which is quite unlike to be accidentally used by a user?

Oh, and why not just `InvalidID`? There is now only one of these constexpr and it's more or less an implementation detail that it is a string.

Reply:
Ideally we want none of these as actual string constants. Do we need to construct it with those? If no (and I think we should not allow it), then we should handle invalidation internally. If we even need invalid ServiceDescriptions, having invalid objects of any kind is almost always a code smell.